### PR TITLE
[RS][DDCE-1503] Bootstrap & Play 2.7 upgrade.

### PR DESCRIPTION
**Pull request by RafSobol**
# DDCE-1503

Bootstrap & Play stuff.
Acceptance tests currently intermittently failing because of strange behaviour of the stubs post-play update.

## Checklist PR Raiser
- [x]  I've made every effort to commit high quality, clean code and I have executed relevant static analyses to be sure
- [x]  I've included appropriate tests with any code I've added (Unit, Integration, Acceptance etc.)
- [x]  I've executed the acceptance test pack locally to ensure there are no functional regressions (Performance tests where relevant)
- [x]  I've squashed my commits - including the JIRA issue number in the commit message
- [x]  I've run a dependency check to ensure all dependencies are up to date

### CODE REVIEW

**Comment by RafSobol** [comment]
the stub issue is resolved, this is ready for review

File: build.sbt [hunk 1]
```diff
@@ -9,7 +9,7 @@ import uk.gov.hmrc.versioning.SbtGitVersioning.autoImport.majorVersion
 
 val appName = "birth-registration-matching"
 
-lazy val plugins : Seq[Plugins] = Seq(play.sbt.PlayScala, SbtAutoBuildPlugin, SbtGitVersioning, SbtDistributablesPlugin, SbtArtifactory)
+lazy val plugins : Seq[Plugins] = Seq(play.sbt.PlayScala, SbtGitVersioning, SbtDistributablesPlugin, SbtArtifactory)
```
Comment (DannyPennington97) [inline build.sbt]: SbtGitVersioning and SbtArtifactory can be removed here (auto-build handles them)

File: project/AppDependencies.scala [hunk 2]
```diff
@@ -3,19 +3,24 @@ import sbt.{ModuleID, _}
 
 object AppDependencies {
 
+  val silencerVersion = "1.7.1"
+
   val compile: Seq[ModuleID] = Seq(
-    "uk.gov.hmrc" %% "bootstrap-play-26" % "2.3.0",
-    "uk.gov.hmrc" %% "play-ui" % "8.19.0-play-26"
+    "uk.gov.hmrc" %% "bootstrap-backend-play-27" % "5.3.0",
+    "uk.gov.hmrc" %% "play-ui" % "9.2.0-play-27",
```
Comment (DannyPennington97) [inline project/AppDependencies.scala]: I feel like we don't need play-ui as an api? Should be fine to remove

File: build.sbt (hunk 1 above)
Comment (RafSobol) [reply build.sbt]: will get rid of them

File: project/plugins.sbt [hunk 3]
```diff
@@ -2,15 +2,13 @@ resolvers += "Typesafe Releases" at "https://repo.typesafe.com/typesafe/releases
...
 resolvers += Resolver.url("HMRC-open-artefacts-ivy", url("https://open.artefacts.tax.service.gov.uk/ivy2"))(Resolver.ivyStylePatterns)
 
-addSbtPlugin("uk.gov.hmrc" % "sbt-auto-build" % "2.15.0")
-
-addSbtPlugin("uk.gov.hmrc" % "sbt-git-versioning" % "2.2.0")
+addSbtPlugin("uk.gov.hmrc" % "sbt-auto-build" % "3.0.0")
 
 addSbtPlugin("uk.gov.hmrc" % "sbt-distributables" % "2.1.0")
 
 addSbtPlugin("uk.gov.hmrc" % "sbt-artifactory" % "1.13.0")
```
Comment (DannyPennington97) [inline project/plugins.sbt]: sbt-artifactory can be removed here too

File: project/AppDependencies.scala (hunk 2 above)
Comment (RafSobol) [reply project/AppDependencies.scala]: frankly I forgot to check, I kinda assumed it's using some helper methods from it or some such but I just tested and it indeed is unneeded - will toss it out

**Comment by DannyPennington97** [comment]
It's not something you've changed but there's an `it` folder without any integration tests... Can you remove it (and also all the integration test stuff in the build.sbt)

**Comment by RafSobol** [comment]
> It's not something you've changed but there's an `it` folder without any integration tests... Can you remove it (and also all the integration test stuff in the build.sbt)

with great pleasure. I'll take this as an excuse to tidy up the build.sbt

**Review by DannyPennington97 — APPROVED** [review]

**Pull request merged**